src/main/java/edu/kis/vh/nursery/DefaultCountingOutRhymer.java

```diff
@@ -1,33 +1,78 @@
 package edu.kis.vh.nursery;
 
+/**
+ * Basic rhymer type implementation
+ */
 public class DefaultCountingOutRhymer {
 
-    private int[] NUMBERS = new int[12];
-    public int total = -1;
+    private static final int STACK_CAPACITY = 12;
+    private static final int ERROR_CODE = -1;
+    private static final int MAX_SIZE = 11;
+    private static final int EMPTY_POSITION = -1;
 
-    public void countIn(int in) {
+    private int total = getEmptyPosition();
+    private final int[] NUMBERS = new int[getStackCapacity()];
+
+    public static int getStackCapacity() {

```

Comment:
Constants usually shouldn't be encapsulated. This is an exaggeration. Moreover, using private fields internally usually do not require encapsulation.